### PR TITLE
[LENS-579] Use date-fns to format date strings

### PR DESCRIPTION
**Pull request by dbchristopher**
### :sparkles: Changes

- Updates DateTimeFormat to use date-fns more reliable formatting and timezone conversion functions (rather than built-in browser default)
- Split documentation for DateFormat and TimeFormat into two different pages; remove references to DateTimeFormat as it's not the preferred public-facing API 

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] Update documentation
- [x] Updated CHANGELOG
- [x] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [x] PR is ideally < ~400LOC

### CODE REVIEW

File: packages/www/src/documentation/components/content/date-time-format.mdx [hunk 1]
```diff
@@ -1,74 +0,0 @@
----
-title: DateTimeFormat
-propsOf: DateTimeFormat
-github: DateTimeFormat
----
```
Comment (dbchristopher) [inline packages/www/src/documentation/components/content/date-time-format.mdx:5]: I split this documentation into two separate pages for `DateFormat` and `TimeFormat`. It seems the simplest to try to keep one component per mdx file where possible

File: packages/components/package.json [hunk 2]
```diff
@@ -24,6 +24,7 @@
     "d3-color": "^1.4.1",
     "d3-hsv": "^0.1.0",
     "date-fns": "^2.12.0",
+    "date-fns-tz": "^1.0.10",
```
Comment (lukelooker) [inline packages/components/package.json:27]: Neat!

File: packages/components/src/DateTimeFormat/DateTimeFormat.tsx [hunk 3]
```diff
@@ -25,13 +25,16 @@
...
+import trim from 'lodash/trim'
+import { LocaleCodes, Locales, formatDateString } from '../utils'
 
 type DateFormats = 'short' | 'medium' | 'long' | 'full'
 
 export interface DateTimeFormatProps {
   children?: Date
   format?: DateFormats
-  locale?: string
+  locale?: LocaleCodes
```
Comment (lukelooker) [inline packages/components/src/DateTimeFormat/DateTimeFormat.tsx:37]: Huzzah!

File: CHANGELOG.md [hunk 4]
```diff
@@ -18,6 +18,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - Doing this means that tooltip trigger elements can now have an `aria-describedby` property with said id as the value
 - `MenuDisclosure`, `Banner`, `IconButton`, `ModalHeader` explicitly use their id props to either provide `useTooltip` with an id or to provide another component that uses `useTooltip` with an id to generate the tooltip's id
 - `Banner` fontSize adjusted and external margin removed
+- `DateTimeFormat` uses date-fns to format human-readable date string (rather than built-in browser default functionality)
+
+## [RELEASED]
+
+### Changed
+
```
Comment (lukelooker) [inline CHANGELOG.md:26]: I think this is spurious (boy am I tired of CHANGELOG conflicts) :/

```suggestion
```

**Comment by dbchristopher** [comment]
Yeah I think localization is big enough to warrant its own section! I'm sure it would be helpful to collect all our tips and resources around that

**Pull request merged**